### PR TITLE
Manually merged release-0.4.1 branch and 27-new-caching-solution.

### PR DESCRIPTION
**Pull request by mpetojevic**
`27-react-query-caching` contains all changes from [27-new-caching-solution](https://github.com/TU-Wien-dataLAB/Grader-Labextension/tree/27-new-caching-solution). I had to manually merge my old branch with `release-0.4.1` due to modifications in file selection for pushing, not present in the original `27-new-caching-solution` branch, leading to many conflicts. In the  `27-react-query-caching`changes from `release-0.4.1`  and `27-new-caching-solution` are integrated.

Ref: #27

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,7 +73,8 @@
         "@mui/material": "^5.13.4",
         "@mui/system": "^5.13.2",
         "@mui/x-date-pickers": "^6.19.0",
-        "@tanstack/react-query": "^5.20.5",
+        "@tanstack/react-query": "^5.27.5",
+        "@tanstack/react-query-devtools": "^5.27.8",
```
Comment (meffmadd) [inline package.json:77]: react-query-devtools hört sich nach dev dependency an

File: src/components/coursemanage/grading/edit-submission.tsx [hunk 2]
```diff
@@ -234,12 +237,21 @@ export const EditSubmission = () => {
...
-            sx={{ fontSize: 10, fontFamily: "'Roboto Mono', monospace" }}
-          >
-            {logs}
-          </Typography>
+          {logs ? (
+            <Typography
+              id="alert-dialog-description"
+              sx={{ fontSize: 10, fontFamily: "'Roboto Mono', monospace" }}
+            >
+              {logs}
```
Comment (meffmadd) [inline src/components/coursemanage/grading/edit-submission.tsx:245]: Könnte man umbschrieben in `{logs | "No logs available"}` dann spart man sich den zweiten Typography tag

**Review by meffmadd — CHANGES_REQUESTED** [review]
The implementation looks very clean!

2 points:
- I don't think caching is configured correctly. The reason we wanted to switch to Tanstack Query is to have client-side caching. When I go to the coursemanage view, click on a lecture, go back, click on the same lecture again, etc., there are always requests made through the request service. I think the `staleTime` has to be set to be `>0`?
- We can then remove the current HTTP browser-based caching by removing the `@cache` decorator from the tornado handler functions.

**Review by florian-jaeger — CHANGES_REQUESTED** [review]
Good job! I experienced the redundant data fetching as well that @meffmadd pointed out.

**Pull request merged**